desc/compute/_stability.py

```diff
@@ -398,11 +398,27 @@ def _ideal_ballooning_gamma2(params, transforms, profiles, data, **kwargs):
 
     A_redo = B_inv @ A @ jnp.transpose(B_inv, axes=(0, 1, 3, 2))
 
-    w, _ = jnp.linalg.eigh(A_redo)
-    # max over "zeta" axis, still a function of rho, alpha, zeta0
-    gamma = jnp.real(jnp.max(w, axis=(2,)))
+    # TODO: Issue #1750
+    # Try jax.scipy.eigh_tridiagonal for increased performance
+    w, v = jnp.linalg.eigh(A_redo)
 
-    data["ideal ballooning lambda"] = gamma.flatten()
+    flat_w = w.flatten()
+
+    top_eigvals, top_flat_indices = jax.lax.top_k(flat_w, k=N_zeta - 2)
+
+    # convert back to original shape
+    alpha_idx, zeta0_idx, theta_idx = jnp.unravel_index(top_flat_indices, w.shape)
+
+    # v becomes less than the precision at some theta points which gives NaNs
+    # stop_gradient prevents that. Not sure how it will affect an objective that
+    # requires both the eigenvalue and eigenfunction
+    top_eigenfuns = jnp.transpose(jax.lax.stop_gradient(v), axes=(0, 1, 3, 2))[
+        alpha_idx, zeta0_idx, theta_idx
+    ]
+    # Combine eigenvalue and eigenfunction data (Neigvals, Ntheta-1)
+    # Eigenvalues are on the top row
+    top_eigvals_and_eigfuns = jnp.concat((top_eigvals[:, None], top_eigenfuns), axis=1)
+    data["ideal ballooning lambda"] = top_eigvals_and_eigfuns.flatten()
```

Comment:
would be useful to include a comment here to remind us what the "natural" shape of this is since its not the usual (nrho, ntheta, nzeta). AKA, if we want to reshape it to separate eigenvalues/eigenvectors by fieldline or by zeta etc what is the correct `reshape`